lib/counter_culture/reconciler.rb

```diff
@@ -19,123 +19,161 @@ def reconcile!
       return false if @reconciled
 
       if options[:skip_unsupported]
-        return false if (foreign_key_values || (counter_cache_name.is_a?(Proc) && !column_names) || delta_magnitude.is_a?(Proc))
+        return false if (foreign_key_values || (counter_cache_name.is_a?(Proc) && !column_names) || delta_magnitude.is_a?(Proc) || (polymorphic? && polymorphic_associated_models.nil?))
       else
         raise "Fixing counter caches is not supported when using :foreign_key_values; you may skip this relation with :skip_unsupported => true" if foreign_key_values
         raise "Must provide :column_names option for relation #{relation.inspect} when :column_name is a Proc; you may skip this relation with :skip_unsupported => true" if counter_cache_name.is_a?(Proc) && !column_names
         raise "Fixing counter caches is not supported when :delta_magnitude is a Proc; you may skip this relation with :skip_unsupported => true" if delta_magnitude.is_a?(Proc)
+        raise "Fixing counter caches is not supported on polymorphic associations unless you specify `polymorphic_associated_models: ['Model1', 'Model2']` on the counter_cache" if polymorphic? && polymorphic_associated_models.nil?
       end
 
-      # if we're provided a custom set of column names with conditions, use them; just use the
-      # column name otherwise
-      # which class does this relation ultimately point to? that's where we have to start
+      if polymorphic?
+        polymorphic_associated_models.each do |model|
```

Comment:
Ok I hadn't thought of that. I'll see if it works throughout and we can ditch that option.